src/pipeline_migration/actions/add_task.py

```diff
@@ -0,0 +1,406 @@
+            logger.info(
+                "Task %s is being referenced in pipeline %s already.",
+                actual_task_name,
+                pipeline_file,
+            )
+            continue
+
+        if (
+            pipeline_task_name in existing_actual_task_names
+            or actual_task_name in existing_pipeline_task_names
+        ):
+            logger.warning(
+                "The pipeline task name and actual task name seem swapped. Skip adding task."
+            )
+            continue
+
+        tasks.append(task_config)
+        dump_yaml(pipeline_file, doc, style)
+        logger.info("Task %s is added to pipeline %s", actual_task_name, original_abs_path)
+
+        if args.git_add:
+            git_add(original_abs_path)
+            logger.info("%s is added to git index.", original_abs_path)
+
+
+def git_add(file_path: FilePath) -> None:
+    """Git add given file
+
+    The git-add command may fail due to any reason, e.g. git command is not available in the system,
+    in which case just logging a message and terminate quietly.
+
+    :param file_path: an absolute path to a file.
+    :type file_path: FilePath
+    :raises ValueError: if given file path is not an absolute path.
+    """
+    fp = Path(file_path)
+    if not fp.is_absolute():
+        raise ValueError(f"File path {file_path} is not an absolute path.")
+    cmd = ["git", "add", fp.name]
+    try:
+        subprocess.run(cmd, cwd=fp.parent, capture_output=True, check=True)
+    except subprocess.CalledProcessError as e:
+        logger.warning("%s is not added to git index: %s", file_path, e.stderr)
+
+
+def search_pipeline_files(files_or_dirs: Iterable[str]) -> Generator[tuple[str, str]]:
+
+    def _iterate_files_or_dirs() -> Generator[Path]:
+        for item in files_or_dirs:
+            if not item:
+                continue
+            entry_path = Path(item).absolute()
+            if entry_path.is_symlink():
+                logger.warning(
+                    "Skip symlink %s. Please specify the destination file or directory instead.",
+                    item,
+                )
+            elif entry_path.is_dir():
+                for entry in entry_path.iterdir():
+                    if entry.is_symlink():
+                        continue
+                    if entry.is_file() and entry.name.endswith(".yaml"):
+                        yield entry
+            elif entry_path.is_file():
+                yield entry_path
+
+    for file_path in _iterate_files_or_dirs():
+        try:
+            with resolve_pipeline(file_path) as pipeline_file:
+                yield str(file_path), str(pipeline_file)
+        except NotAPipelineFile as e:
+            logger.warning("%s is not an expected pipeline file due to: %s", file_path.name, e)
+        except Exception as e:
+            logger.warning("%s seems not a YAML file due to: %s", file_path.name, e)
+
+
+class KonfluxBuildDefinitions:
```

Comment:
This class is a collection of static methods and methods that only have `self` to access the three constants. It might make more sense as a module than a class.

But that's a non-blocking nitpick